AppManager/app_manager_server.py

```diff
@@ -147,6 +147,15 @@ def add_routing(app, port):
     app: A string that contains the application ID.
     port: A string that contains the port that the AppServer listens on.
   """
+  logging.info("Waiting for application {} on port {} to be active.".
+    format(str(app), str(port)))
+  if not wait_on_app(port):
+    # In case the AppServer fails we let the AppController to detect it
```

Comment:
So instead of a MonitInterface.stop() here, we potentially leave the appserver instance behind until the controller stops it?

Is there a possibility Monit would not report the stray server?